Move EKS creation and permissions page from Ref Guides to Setup Cluster Hosted K8s Provider section (#289)
rancher/rancher-docs: docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-clusters-from-hosted-kubernetes-providers/gke.md

## Programmatically Creating GKE Clusters

> The most common way to programmatically deploy GKE clusters through Rancher is by using the Rancher2 Terraform provider. The documentation for creating clusters with Terraform is [here.](https://registry.terraform.io/providers/rancher/rancher2/latest/docs/resources/cluster)

Reviewer: ```suggestion
The most common way to programmatically deploy GKE clusters through Rancher is by using the Rancher2 Terraform provider. See the [official Terraform documentation](https://registry.terraform.io/providers/rancher/rancher2/latest/docs/resources/cluster).
```
